pkgs/test/tool/host.dart

```diff
@@ -173,10 +173,27 @@ MultiChannel<dynamic> _connectToServer() {
   return MultiChannel(controller.foreign);
 }
 
-/// Creates an iframe with `src` [url] and establishes a connection to it using
-/// a [MessageChannel].
+/// Creates an iframe with `src` [url] and expects a message back to connect a
+/// message channel with the suite running in the frame.
 ///
 /// [id] identifies the suite loaded in this iframe.
+///
+/// While the iframe is starting, adds a listener for `window.onMessage` which
```

Comment:
I wonder if this is partly the issue regarding flakyness? If the message posts before our listener is set up we might not see it?

Reply:
It looks easy enough in this implementation to ensure the listener is added before the frame.

Reply:
> I wonder if this is partly the issue regarding flakyness?

Hard to say, but apparently it wasn't the sole issue cause it still flakes on this PR 😝 

https://github.com/dart-lang/test/actions/runs/5717430441/job/15491160863?pr=2068

```
Process `dart bin/test.dart` was killed with SIGKILL in a tear-down. Output:
      00:00 +0: [Microsoft Edge, Dart2Js] loading test.dart
      Info: Compiling with sound null safety.
      Compiled 10,952,326 characters Dart to 1,225,100 characters JavaScript in 7.32 seconds using 327.988 MB of memory
      Dart file ../dart_test_FFBBZA/runInBrowser.dart compiled to JavaScript: ../dart_test_VZICMC/test_AFKCDJ/test.dart.browser_test.dart.js
  TimeoutException after 0:00:30.000000: Test timed out after 30 seconds. See https://pub.dev/packages/test#timeouts
```